### DISCUSSION TITLE
Trade-Dangerous in Google Colabs

### DISCUSSION THREAD

**Discussion by jcc10** [eyeonus/Trade-Dangerous#189] [Q&A] [answered]
I'm trying to setup a google colabs notebook for experimenting with fleet carrier trade calculations and I was wondering how one goes about programmatically importing data from the internet.

After I ran `from tradedangerous import ???` I was wondering what I should call to perform the `trade import -P eddblink` command from within python?

(After I get this working and a proof of concept, I'll share my work for the class :) )

**Comment by eyeonus** [comment]
You have to import the plugin at the least and call its run method with the arguments needed.

Also see https://github.com/eyeonus/Trade-Dangerous/wiki/Python-Quick-Peek

**Reply by jcc10** [reply, marked as answer]
Currently, I am using the following:

```python3
from tradedangerous import tradedb, cli
cli.trade(["", "import", "-P", "eddblink", "-O", "clean"])
```

This is in-fact not described in the python quick peek, but is required to use the script in google colab since you can't get a shell without paying extra.

I'm working on a colab notebook for system level trade routing as a proof of concept. I'll post it after I verify the basics work.

**Reply by eyeonus** [reply]
No, the linked is intended as an introduction to the python illiterate, but the examples it contains should be enough for the persistent to understand the mechanics of using TD as an API.

To use an import plugin, you must import the plugin(s) you want to use, the plugin module, and any modules they depend on.

For example, the server listener's TD imports:
``` python
    from tradedangerous import cli as trade, cache, tradedb, tradeenv, transfers, plugins, commands
    from tradedangerous.plugins import eddblink_plug, spansh_plug
```
